### DISCUSSION TITLE
At startup : audio lost

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/grandorgue#750] [Help]
I don’t know if it is an issue as #731 but when I start the computer and GO at computer startup, when the organ is loaded there is no sound. 

To resolve, I have to go to the Audio Settings, deselect Pa and Jack, apply this settings, then come back to the Audio setting, select Pa and Jack and the sound comes back. 

I tried just to deselect Pa or Jack, but it works only if the two parameters are deselected and reselected

**Comment by oleg68** [comment]
It is not related to #731

Do you have any error messages at startup?

What audio device with what port is selected in Audio Settings?

Does the sound appear if you just select `Panic` from the menu?

**Reply by Jean-Gabriel1982** [reply]
> It is not related to #731
> 
> Do you have any error messages at startup?
> 
> What audio device with what port is selected in Audio Settings?
> 
> Does the sound appear if you just select `Panic` from the menu?

I don’t have any error message at startup. 
if I clic on Panic the sound does not appear. 
![Uploading image.jpg…]()

**Comment by oleg68** [comment]
The image is not visible

**Reply by Jean-Gabriel1982** [reply]
Sorry. 
here it is …
![image](https://user-images.githubusercontent.com/20370038/135730033-f9b6baca-b51d-4105-8709-a12c7a13448d.jpg)

**Comment by oleg68** [comment]
Could you change the output device to Rt?

**Reply by Jean-Gabriel1982** [reply]
I have an error at startup 
Error RtAudio:RtApiAlsa::probeDeviceOpen: pcm device (hw:2,0) won’t open for output

I selected Rt: Alsa :hw:HD-Audio Generic,0:
Thé RtAlsa default didn’t work neither the Generic 1

**Reply by oleg68** [reply]
Which audio devices there are in the change fialog?

**Reply by Jean-Gabriel1982** [reply]
> Which audio devices there are in the change fialog?

i have : 
![image](https://user-images.githubusercontent.com/20370038/135741198-ec5adecc-78a5-4832-b8de-fa31986474a6.jpg)

**Comment by oleg68** [comment]
Try to use ``PA: ALSA: pulse`` and ``Rt: alsa: default``

**Reply by Jean-Gabriel1982** [reply]
It seems to be ok for Pulse at startup. 
i needed to restart completely. If I put Pulse and apply it didn’t work. 

Then I select Pulse, I completely restarted my computer and it seems to be ok now

**Comment by midimusic** [comment]
This is a typical problem which I get as well with Ubuntu. The audio out is all a bit random. Most of the problems derive from not fully understanding the audio system and how pulse, jack and ALSA interact. I have never been able to find a decent explanation (with a block diagram of the sound system) indicating how these three interact. Jack is the usual suspect, which, even when needed, may not be running.
There is also the problem of numerous audio inputs and outputs which get listed even when they do not exist.

"Could do better"

csw900

**Reply by Jean-Gabriel1982** [reply]
Yes I confirm ... it does not work anymore and I'm on Ubuntu ... I think i'm lucky :D

**Comment by oleg68** [comment]
>  Most of the problems derive from not fully understanding the audio system and how pulse, jack and ALSA interact. I have never been able to find a decent explanation (with a block diagram of the sound system) indicating how these three interact.

[Pipewire](https://pipewire.org/) comes as a replacement of both pulse and jack. It has less problems with running GO.

With pulse, it is easy to run GrandOrgue with a separate sound card. It always works.

Otherwise GrandOrgue and pulse share the same audio card with lots of issues.
